File: PST/FUEL Source Code/FUEL/Backup/frmZipProgress.cs
// Decompiled with JetBrains decompiler
// Type: FUEL.frmZipProgress
// Assembly: FUEL, Version=2.4.8.0, Culture=neutral, PublicKeyToken=null
// MVID: A3D630D0-0620-425D-8CFB-B67B2AC75507
// Assembly location: C:\hp\FUEL\FUEL.dll

using DevComponents.DotNetBar;
using DevComponents.DotNetBar.Controls;
using DevComponents.DotNetBar.Metro;
using FUEL.My;
using Ionic.Zip;
using Microsoft.VisualBasic.CompilerServices;
using Microsoft.VisualBasic.FileIO;
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.ComponentModel;
using System.Diagnostics;
using System.Drawing;
using System.IO;
using System.Runtime.CompilerServices;
using System.Windows.Forms;

namespace FUEL
{
  [DesignerGenerated]
  public class frmZipProgress : MetroForm
  {
    private static List<WeakReference> __ENCList = new List<WeakReference>();
    private IContainer components;
    [AccessedThroughProperty("prgMain")]
    private ProgressBarX _prgMain;
    [AccessedThroughProperty("prgAlternate")]
    private ProgressBarX _prgAlternate;
    [AccessedThroughProperty("TableLayoutPanel1")]
    private TableLayoutPanel _TableLayoutPanel1;
    [AccessedThroughProperty("lblFile")]
    private LabelX _lblFile;
    [AccessedThroughProperty("lblJob")]
    private LabelX _lblJob;

    [DebuggerNonUserCode]
    static frmZipProgress()
    {
    }

    [DebuggerNonUserCode]
    private static void __ENCAddToList(object value)
    {
      lock (frmZipProgress.__ENCList)
      {
        if (frmZipProgress.__ENCList.Count == frmZipProgress.__ENCList.Capacity)
        {
          int index1 = 0;
          int num = checked (frmZipProgress.__ENCList.Count - 1);
          int index2 = 0;
          while (index2 <= num)
          {
            if (frmZipProgress.__ENCList[index2].IsAlive)
            {
              if (index2 != index1)
                frmZipProgress.__ENCList[index1] = frmZipProgress.__ENCList[index2];
              checked { ++index1; }
            }
            checked { ++index2; }
          }
          frmZipProgress.__ENCList.RemoveRange(index1, checked (frmZipProgress.__ENCList.Count - index1));
          frmZipProgress.__ENCList.Capacity = frmZipProgress.__ENCList.Count;
        }
        frmZipProgress.__ENCList.Add(new WeakReference(RuntimeHelpers.GetObjectValue(value)));
      }
    }

    [DebuggerNonUserCode]
    protected override void Dispose(bool disposing)
    {
      try
      {
        if (!disposing || this.components == null)
          return;
        this.components.Dispose();
      }
      finally
      {
        base.Dispose(disposing);
      }
    }

    [DebuggerStepThrough]
    private void InitializeComponent()
    {
      this.prgMain = new ProgressBarX();
      this.prgAlternate = new ProgressBarX();
      this.TableLayoutPanel1 = new TableLayoutPanel();
      this.lblFile = new LabelX();
      this.lblJob = new LabelX();
      this.TableLayoutPanel1.SuspendLayout();
      this.SuspendLayout();
      this.prgMain.BackColor = Color.White;
      this.prgMain.BackgroundStyle.CornerType = eCornerType.Square;
      this.prgMain.Dock = DockStyle.Fill;
      this.prgMain.ForeColor = Color.Black;
      ProgressBarX prgMain1 = this.prgMain;
      Point point1 = new Point(3, 95);
      Point point2 = point1;
      prgMain1.Location = point2;
      this.prgMain.Name = "prgMain";
      ProgressBarX prgMain2 = this.prgMain;
      Size size1 = new Size(363, 26);
      Size size2 = size1;
      prgMain2.Size = size2;
      this.prgMain.Step = 0;
      this.prgMain.TabIndex = 0;
      this.prgMain.Text = "ProgressBarX1";
      this.prgAlternate.BackColor = Color.White;
      this.prgAlternate.BackgroundStyle.CornerType = eCornerType.Square;
      this.prgAlternate.Dock = DockStyle.Fill;
      this.prgAlternate.ForeColor = Color.Black;
      ProgressBarX prgAlternate1 = this.prgAlternate;
      point1 = new Point(3, 33);
      Point point3 = point1;
      prgAlternate1.Location = point3;
      this.prgAlternate.Name = "prgAlternate";
      ProgressBarX prgAlternate2 = this.prgAlternate;
      size1 = new Size(363, 26);
      Size size3 = size1;
      prgAlternate2.Size = size3;
      this.prgAlternate.TabIndex = 1;
      this.prgAlternate.Text = "ProgressBarX1";
      this.TableLayoutPanel1.BackColor = Color.FromArgb(211, 211, 211);
      this.TableLayoutPanel1.ColumnCount = 1;
      this.TableLayoutPanel1.ColumnStyles.Add(new ColumnStyle(SizeType.Percent, 100f));
      this.TableLayoutPanel1.Controls.Add((Control) this.prgMain, 0, 3);
      this.TableLayoutPanel1.Controls.Add((Control) this.prgAlternate, 0, 1);
      this.TableLayoutPanel1.Controls.Add((Control) this.lblFile, 0, 0);
      this.TableLayoutPanel1.Controls.Add((Control) this.lblJob, 0, 2);
      this.TableLayoutPanel1.Dock = DockStyle.Fill;
      this.TableLayoutPanel1.ForeColor = Color.Black;
      TableLayoutPanel tableLayoutPanel1_1 = this.TableLayoutPanel1;
      point1 = new Point(0, 0);
      Point point4 = point1;
      tableLayoutPanel1_1.Location = point4;
      this.TableLayoutPanel1.Name = "TableLayoutPanel1";
      this.TableLayoutPanel1.RowCount = 4;
      this.TableLayoutPanel1.RowStyles.Add(new RowStyle(SizeType.Absolute, 30f));
      this.TableLayoutPanel1.RowStyles.Add(new RowStyle(SizeType.Percent, 50f));
      this.TableLayoutPanel1.RowStyles.Add(new RowStyle(SizeType.Absolute, 30f));
      this.TableLayoutPanel1.RowStyles.Add(new RowStyle(SizeType.Percent, 50f));
      TableLayoutPanel tableLayoutPanel1_2 = this.TableLayoutPanel1;
      size1 = new Size(369, 124);
      Size size4 = size1;
      tableLayoutPanel1_2.Size = size4;
      this.TableLayoutPanel1.TabIndex = 2;
      this.lblFile.BackColor = Color.Transparent;
      this.lblFile.BackgroundStyle.CornerType = eCornerType.Square;
      this.lblFile.ForeColor = Color.Black;
      LabelX lblFile1 = this.lblFile;
      point1 = new Point(3, 3);
      Point point5 = point1;
      lblFile1.Location = point5;
      this.lblFile.Name = "lblFile";
      LabelX lblFile2 = this.lblFile;
      size1 = new Size(75, 23);
      Size size5 = size1;
      lblFile2.Size = size5;
      this.lblFile.TabIndex = 2;
      this.lblFile.Text = "File Progress";
      this.lblJob.BackColor = Color.Transparent;
      this.lblJob.BackgroundStyle.CornerType = eCornerType.Square;
      this.lblJob.ForeColor = Color.Black;
      LabelX lblJob1 = this.lblJob;
      point1 = new Point(3, 65);
      Point point6 = point1;
      lblJob1.Location = point6;
      this.lblJob.Name = "lblJob";
      LabelX lblJob2 = this.lblJob;
      size1 = new Size(75, 23);
      Size size6 = size1;
      lblJob2.Size = size6;
      this.lblJob.TabIndex = 3;
      this.lblJob.Text = "Job Progress";
      this.AutoScaleDimensions = new SizeF(6f, 13f);
      this.AutoScaleMode = AutoScaleMode.Font;
      size1 = new Size(369, 124);
      this.ClientSize = size1;
      this.ControlBox = false;
      this.Controls.Add((Control) this.TableLayoutPanel1);
      this.DoubleBuffered = true;
      this.Font = new Font("Segoe UI", 8.25f, FontStyle.Regular, GraphicsUnit.Point, (byte) 0);
      this.FormBorderStyle = FormBorderStyle.FixedToolWindow;
      this.Name = nameof (frmZipProgress);
      this.ShowIcon = false;
      this.ShowInTaskbar = false;
      this.TableLayoutPanel1.ResumeLayout(false);
      this.ResumeLayout(false);
    }

    internal virtual ProgressBarX prgMain
    {
      [DebuggerNonUserCode] get => this._prgMain;
      [DebuggerNonUserCode, MethodImpl(MethodImplOptions.Synchronized)] set => this._prgMain = value;
    }

    internal virtual ProgressBarX prgAlternate
    {
      [DebuggerNonUserCode] get => this._prgAlternate;
      [DebuggerNonUserCode, MethodImpl(MethodImplOptions.Synchronized)] set => this._prgAlternate = value;
    }

    internal virtual TableLayoutPanel TableLayoutPanel1
    {
      [DebuggerNonUserCode] get => this._TableLayoutPanel1;
      [DebuggerNonUserCode, MethodImpl(MethodImplOptions.Synchronized)] set => this._TableLayoutPanel1 = value;
    }

    internal virtual LabelX lblFile
    {
      [DebuggerNonUserCode] get => this._lblFile;
      [DebuggerNonUserCode, MethodImpl(MethodImplOptions.Synchronized)] set => this._lblFile = value;
    }

    internal virtual LabelX lblJob
    {
      [DebuggerNonUserCode] get => this._lblJob;
      [DebuggerNonUserCode, MethodImpl(MethodImplOptions.Synchronized)] set => this._lblJob = value;
    }

    private string _SourceDir { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    private string _OutputName { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    private long _BytesTransfered { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    private long _JobSize { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    private long _MaxFileSize { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    private bool _DeleteFilesAfterZip { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    private bool _HideForm { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    public long MaxFileSize
    {
      get => this._MaxFileSize;
      set => this._MaxFileSize = checked (value * 1024L * 1024L);
    }

    public bool DeleteFilesAfterZip
    {
      get => this._DeleteFilesAfterZip;
      set => this._DeleteFilesAfterZip = value;
    }

    public bool HideForm
    {
      get => this._HideForm;
      set => this._HideForm = value;
    }

    public frmZipProgress(string SourceDir, string OutputName, bool DeleteSourceFiles)
    {
      this.Shown += new EventHandler(this.frmSimpleProgress_Shown);
      frmZipProgress.__ENCAddToList((object) this);
      this._BytesTransfered = 0L;
      this._JobSize = 0L;
      this._MaxFileSize = 0L;
      this._DeleteFilesAfterZip = false;
      this._HideForm = false;
      this.InitializeComponent();
      this._SourceDir = SourceDir;
      this._OutputName = OutputName;
      this._DeleteFilesAfterZip = DeleteSourceFiles;
      ReadOnlyCollection<string> files = MyProject.Computer.FileSystem.GetFiles(this._SourceDir, Microsoft.VisualBasic.FileIO.SearchOption.SearchAllSubDirectories, (string[]) null);
      long num;
      try
      {
        foreach (string file in files)
          checked { num += MyProject.Computer.FileSystem.GetFileInfo(file).Length; }
      }
      finally
      {
        IEnumerator<string> enumerator;
        enumerator?.Dispose();
      }
      this._JobSize = num;
    }

    private void frmSimpleProgress_Shown(object sender, EventArgs e)
    {
      if (this._HideForm)
        this.Hide();
      this.Zip();
      if (!this.DeleteFilesAfterZip)
        return;
      this.DeleteSourceFiles();
    }

    private void DeleteSourceFiles()
    {
      ReadOnlyCollection<string> files = MyProject.Computer.FileSystem.GetFiles(this._SourceDir, Microsoft.VisualBasic.FileIO.SearchOption.SearchAllSubDirectories, (string[]) null);
      try
      {
        foreach (string str in files)
        {
          Path.GetExtension(str);
          if (Operators.CompareString(Path.GetFileNameWithoutExtension(str), Path.GetFileNameWithoutExtension(this._OutputName), false) != 0)
            MyProject.Computer.FileSystem.DeleteFile(str, UIOption.OnlyErrorDialogs, RecycleOption.DeletePermanently, UICancelOption.DoNothing);
        }
      }
      finally
      {
        IEnumerator<string> enumerator;
        enumerator?.Dispose();
      }
      ReadOnlyCollection<string> directories = MyProject.Computer.FileSystem.GetDirectories(this._SourceDir, Microsoft.VisualBasic.FileIO.SearchOption.SearchAllSubDirectories);
      try
      {
        foreach (string directory in directories)
          MyProject.Computer.FileSystem.DeleteDirectory(directory, DeleteDirectoryOption.DeleteAllContents);
      }
      finally
      {
        IEnumerator<string> enumerator;
        enumerator?.Dispose();
      }
    }

    private void Zip()
    {
      try
      {
        using (ZipFile zipFile = new ZipFile())
        {
          zipFile.AddDirectory(this._SourceDir);
          if (this.MaxFileSize != 0L)
            zipFile.MaxOutputSegmentSize = checked ((int) this.MaxFileSize);
          zipFile.SaveProgress += new EventHandler<SaveProgressEventArgs>(this.Zip_SaveProgress);
          zipFile.Save(this._OutputName);
        }
      }
      catch (Exception ex)
      {
        ProjectData.SetProjectError(ex);
        this.Show();
        ProjectData.ClearProjectError();
      }
    }

    private void Zip_SaveProgress(object sender, SaveProgressEventArgs e)
    {
      switch (e.EventType)
      {
        case ZipProgressEventType.Saving_AfterWriteEntry:
          this.StepArchiveProgress(e);
          break;
        case ZipProgressEventType.Saving_Completed:
          this.SaveCompleted();
          break;
        case ZipProgressEventType.Saving_EntryBytesRead:
          this.StepEntryProgress(e);
          break;
      }
    }

    private void StepArchiveProgress(SaveProgressEventArgs e)
    {
      if (this.prgMain.InvokeRequired)
      {
        this.prgMain.Invoke((Delegate) new frmZipProgress.SaveEntryProgress(this.StepArchiveProgress), (object) e);
      }
      else
      {
        checked { this._BytesTransfered += e.CurrentEntry.UncompressedSize; }
        this.prgMain.Value = checked ((int) Math.Round(Math.Round(unchecked ((double) this._BytesTransfered / (double) this._JobSize * 100.0), 0)));
        this.Update();
      }
    }

    private void SaveCompleted() => this.Close();

    private void StepEntryProgress(SaveProgressEventArgs e)
    {
      if (this.prgAlternate.InvokeRequired)
      {
        this.prgAlternate.Invoke((Delegate) new frmZipProgress.SaveEntryProgress(this.StepEntryProgress), (object) e);
      }
      else
      {
        this.prgAlternate.Value = checked ((int) Math.Round(Math.Round(unchecked ((double) e.BytesTransferred / (double) e.TotalBytesToTransfer * 100.0), 0)));
        this.prgMain.Value = checked ((int) Math.Round(Math.Round(unchecked ((double) checked (this._BytesTransfered + e.BytesTransferred) / (double) this._JobSize * 100.0), 0)));
        this.Update();
      }
    }

    private delegate void SaveEntryProgress(SaveProgressEventArgs e);
  }
}

File: PST/FUEL/src/FUEL/frmZipProgress.vb
Imports DevComponents.DotNetBar
Imports DevComponents.DotNetBar.Controls
Imports DevComponents.DotNetBar.Metro
Imports FUEL.My
Imports Ionic.Zip
Imports Microsoft.VisualBasic.CompilerServices
Imports Microsoft.VisualBasic.FileIO
Imports System
Imports System.Collections.Generic
Imports System.Collections.ObjectModel
Imports System.ComponentModel
Imports System.Diagnostics
Imports System.Drawing
Imports System.IO
Imports System.Runtime.CompilerServices
Imports System.Windows.Forms

Namespace FUEL
    <DesignerGenerated> _
    Public Class frmZipProgress
        Inherits MetroForm
        ' Methods
        Public Sub New(ByVal SourceDir As String, ByVal OutputName As String, ByVal DeleteSourceFiles As Boolean)
            Dim num As Long
            Dim enumerator As IEnumerator(Of String)
            AddHandler MyBase.Shown, New EventHandler(AddressOf Me.frmSimpleProgress_Shown)
            frmZipProgress.__ENCAddToList(Me)
            Me._BytesTransfered = 0
            Me._JobSize = 0
            Me._MaxFileSize = 0
            Me._DeleteFilesAfterZip = False
            Me._HideForm = False
            Me.InitializeComponent
            Me._SourceDir = SourceDir
            Me._OutputName = OutputName
            Me._DeleteFilesAfterZip = DeleteSourceFiles
            Dim onlys As ReadOnlyCollection(Of String) = MyProject.Computer.FileSystem.GetFiles(Me._SourceDir, SearchOption.SearchAllSubDirectories, Nothing)
            Try 
                enumerator = onlys.GetEnumerator
                Do While True
                    If Not enumerator.MoveNext Then
                        Exit Do
                    End If
                    Dim current As String = enumerator.Current
                    num = (num + MyProject.Computer.FileSystem.GetFileInfo(current).Length)
                Loop
            Finally
                If Not Object.ReferenceEquals(enumerator, Nothing) Then
                    enumerator.Dispose
                End If
            End Try
            Me._JobSize = num
        End Sub

        <DebuggerNonUserCode> _
        Private Shared Sub __ENCAddToList(ByVal value As Object)
            SyncLock frmZipProgress.__ENCList
                If (frmZipProgress.__ENCList.Count = frmZipProgress.__ENCList.Capacity) Then
                    Dim index As Integer = 0
                    Dim num3 As Integer = (frmZipProgress.__ENCList.Count - 1)
                    Dim num2 As Integer = 0
                    Do While True
                        Dim num4 As Integer = num3
                        If (num2 > num4) Then
                            frmZipProgress.__ENCList.RemoveRange(index, (frmZipProgress.__ENCList.Count - index))
                            frmZipProgress.__ENCList.Capacity = frmZipProgress.__ENCList.Count
                            Exit Do
                        End If
                        Dim reference As WeakReference = frmZipProgress.__ENCList(num2)
                        If reference.IsAlive Then
                            If (num2 <> index) Then
                                frmZipProgress.__ENCList(index) = frmZipProgress.__ENCList(num2)
                            End If
                            index += 1
                        End If
                        num2 += 1
                    Loop
                End If
                frmZipProgress.__ENCList.Add(New WeakReference(value))
            End SyncLock
        End Sub

        Private Sub DeleteSourceFiles()
            Dim enumerator As IEnumerator(Of String)
            Dim enumerator2 As IEnumerator(Of String)
            Dim onlys2 As ReadOnlyCollection(Of String) = MyProject.Computer.FileSystem.GetFiles(Me._SourceDir, SearchOption.SearchAllSubDirectories, Nothing)
            Try 
                enumerator = onlys2.GetEnumerator
                Do While True
                    If Not enumerator.MoveNext Then
                        Exit Do
                    End If
                    Dim current As String = enumerator.Current
                    Dim extension As String = Path.GetExtension(current)
                    If (Path.GetFileNameWithoutExtension(current) <> Path.GetFileNameWithoutExtension(Me._OutputName)) Then
                        MyProject.Computer.FileSystem.DeleteFile(current, UIOption.OnlyErrorDialogs, RecycleOption.DeletePermanently, UICancelOption.DoNothing)
                    End If
                Loop
            Finally
                If Not Object.ReferenceEquals(enumerator, Nothing) Then
                    enumerator.Dispose
                End If
            End Try
            Dim onlys As ReadOnlyCollection(Of String) = MyProject.Computer.FileSystem.GetDirectories(Me._SourceDir, SearchOption.SearchAllSubDirectories, New String(0  - 1) {})
            Try 
                enumerator2 = onlys.GetEnumerator
                Do While True
                    If Not enumerator2.MoveNext Then
                        Exit Do
                    End If
                    Dim current As String = enumerator2.Current
                    MyProject.Computer.FileSystem.DeleteDirectory(current, DeleteDirectoryOption.DeleteAllContents)
                Loop
            Finally
                If Not Object.ReferenceEquals(enumerator2, Nothing) Then
                    enumerator2.Dispose
                End If
            End Try
        End Sub

        <DebuggerNonUserCode> _
        Protected Overrides Sub Dispose(ByVal disposing As Boolean)
            Try 
                If (If((Not disposing OrElse (Me.components Is Nothing)), 0, 1) <> 0) Then
                    Me.components.Dispose
                End If
            Finally
                MyBase.Dispose(disposing)
            End Try
        End Sub

        Private Sub frmSimpleProgress_Shown(ByVal sender As Object, ByVal e As EventArgs)
            If Me._HideForm Then
                Me.Hide
            End If
            Me.Zip
            If Me.DeleteFilesAfterZip Then
                Me.DeleteSourceFiles
            End If
        End Sub

        <DebuggerStepThrough> _
        Private Sub InitializeComponent()
            Me.prgMain = New ProgressBarX
            Me.prgAlternate = New ProgressBarX
            Me.TableLayoutPanel1 = New TableLayoutPanel
            Me.lblFile = New LabelX
            Me.lblJob = New LabelX
            Me.TableLayoutPanel1.SuspendLayout
            Me.SuspendLayout
            Me.prgMain.BackColor = Color.White
            Me.prgMain.BackgroundStyle.CornerType = eCornerType.Square
            Me.prgMain.Dock = DockStyle.Fill
            Me.prgMain.ForeColor = Color.Black
            Dim point2 As New Point(3, &H5F)
            Me.prgMain.Location = point2
            Me.prgMain.Name = "prgMain"
            Dim size2 As New Size(&H16B, &H1A)
            Me.prgMain.Size = size2
            Me.prgMain.Step = 0
            Me.prgMain.TabIndex = 0
            Me.prgMain.Text = "ProgressBarX1"
            Me.prgAlternate.BackColor = Color.White
            Me.prgAlternate.BackgroundStyle.CornerType = eCornerType.Square
            Me.prgAlternate.Dock = DockStyle.Fill
            Me.prgAlternate.ForeColor = Color.Black
            point2 = New Point(3, &H21)
            Me.prgAlternate.Location = point2
            Me.prgAlternate.Name = "prgAlternate"
            size2 = New Size(&H16B, &H1A)
            Me.prgAlternate.Size = size2
            Me.prgAlternate.TabIndex = 1
            Me.prgAlternate.Text = "ProgressBarX1"
            Me.TableLayoutPanel1.BackColor = Color.FromArgb(&HD3, &HD3, &HD3)
            Me.TableLayoutPanel1.ColumnCount = 1
            Me.TableLayoutPanel1.ColumnStyles.Add(New ColumnStyle(SizeType.Percent, 100!))
            Me.TableLayoutPanel1.Controls.Add(Me.prgMain, 0, 3)
            Me.TableLayoutPanel1.Controls.Add(Me.prgAlternate, 0, 1)
            Me.TableLayoutPanel1.Controls.Add(Me.lblFile, 0, 0)
            Me.TableLayoutPanel1.Controls.Add(Me.lblJob, 0, 2)
            Me.TableLayoutPanel1.Dock = DockStyle.Fill
            Me.TableLayoutPanel1.ForeColor = Color.Black
            point2 = New Point(0, 0)
            Me.TableLayoutPanel1.Location = point2
            Me.TableLayoutPanel1.Name = "TableLayoutPanel1"
            Me.TableLayoutPanel1.RowCount = 4
            Me.TableLayoutPanel1.RowStyles.Add(New RowStyle(SizeType.Absolute, 30!))
            Me.TableLayoutPanel1.RowStyles.Add(New RowStyle(SizeType.Percent, 50!))
            Me.TableLayoutPanel1.RowStyles.Add(New RowStyle(SizeType.Absolute, 30!))
            Me.TableLayoutPanel1.RowStyles.Add(New RowStyle(SizeType.Percent, 50!))
            size2 = New Size(&H171, &H7C)
            Me.TableLayoutPanel1.Size = size2
            Me.TableLayoutPanel1.TabIndex = 2
            Me.lblFile.BackColor = Color.Transparent
            Me.lblFile.BackgroundStyle.CornerType = eCornerType.Square
            Me.lblFile.ForeColor = Color.Black
            point2 = New Point(3, 3)
            Me.lblFile.Location = point2
            Me.lblFile.Name = "lblFile"
            size2 = New Size(&H4B, &H17)
            Me.lblFile.Size = size2
            Me.lblFile.TabIndex = 2
            Me.lblFile.Text = "File Progress"
            Me.lblJob.BackColor = Color.Transparent
            Me.lblJob.BackgroundStyle.CornerType = eCornerType.Square
            Me.lblJob.ForeColor = Color.Black
            point2 = New Point(3, &H41)
            Me.lblJob.Location = point2
            Me.lblJob.Name = "lblJob"
            size2 = New Size(&H4B, &H17)
            Me.lblJob.Size = size2
            Me.lblJob.TabIndex = 3
            Me.lblJob.Text = "Job Progress"
            Dim ef2 As New SizeF(6!, 13!)
            Me.AutoScaleDimensions = ef2
            Me.AutoScaleMode = AutoScaleMode.Font
            size2 = New Size(&H171, &H7C)
            Me.ClientSize = size2
            Me.ControlBox = False
            Me.Controls.Add(Me.TableLayoutPanel1)
            Me.DoubleBuffered = True
            Me.Font = New Font("Segoe UI", 8.25!, FontStyle.Regular, GraphicsUnit.Point, 0)
            Me.FormBorderStyle = FormBorderStyle.FixedToolWindow
            Me.Name = "frmZipProgress"
            Me.ShowIcon = False
            Me.ShowInTaskbar = False
            Me.TableLayoutPanel1.ResumeLayout(False)
            Me.ResumeLayout(False)
        End Sub

        Private Sub SaveCompleted()
            Me.Close
        End Sub

        Private Sub StepArchiveProgress(ByVal e As SaveProgressEventArgs)
            If Me.prgMain.InvokeRequired Then
                Dim args As Object() = New Object() { e }
                Me.prgMain.Invoke(New SaveEntryProgress(AddressOf Me.StepArchiveProgress), args)
            Else
                Me._BytesTransfered = (Me._BytesTransfered + e.CurrentEntry.UncompressedSize)
                Me.prgMain.Value = CInt(Math.Round(Math.Round(CDbl(((CDbl(Me._BytesTransfered) / CDbl(Me._JobSize)) * 100)), 0)))
                MyBase.Update
            End If
        End Sub

        Private Sub StepEntryProgress(ByVal e As SaveProgressEventArgs)
            If Me.prgAlternate.InvokeRequired Then
                Dim args As Object() = New Object() { e }
                Me.prgAlternate.Invoke(New SaveEntryProgress(AddressOf Me.StepEntryProgress), args)
            Else
                Me.prgAlternate.Value = CInt(Math.Round(Math.Round(CDbl(((CDbl(e.BytesTransferred) / CDbl(e.TotalBytesToTransfer)) * 100)), 0)))
                Me.prgMain.Value = CInt(Math.Round(Math.Round(CDbl(((CDbl((Me._BytesTransfered + e.BytesTransferred)) / CDbl(Me._JobSize)) * 100)), 0)))
                MyBase.Update
            End If
        End Sub

        Private Sub Zip()
            Try 
                Dim objA As New ZipFile
                Try 
                    objA.AddDirectory(Me._SourceDir)
                    If (Me.MaxFileSize <> 0) Then
                        objA.MaxOutputSegmentSize = CInt(Me.MaxFileSize)
                    End If
                    AddHandler objA.SaveProgress, New EventHandler(Of SaveProgressEventArgs)(AddressOf Me.Zip_SaveProgress)
                    objA.Save(Me._OutputName)
                Finally
                    If Not Object.ReferenceEquals(objA, Nothing) Then
                        objA.Dispose
                    End If
                End Try
            Catch exception1 As Exception
                Dim ex As Exception = exception1
                ProjectData.SetProjectError(ex)
                Dim exception As Exception = ex
                Me.Show
                ProjectData.ClearProjectError
            End Try
        End Sub

        Private Sub Zip_SaveProgress(ByVal sender As Object, ByVal e As SaveProgressEventArgs)
            Select Case e.EventType
                Case ZipProgressEventType.Saving_AfterWriteEntry
                    Me.StepArchiveProgress(e)
                    Exit Select
                Case ZipProgressEventType.Saving_Completed
                    Me.SaveCompleted
                    Exit Select
                Case ZipProgressEventType.Saving_EntryBytesRead
                    Me.StepEntryProgress(e)
                    Exit Select
                Case Else
                    Exit Select
            End Select
        End Sub


        ' Properties
        Friend Overridable Property prgMain As ProgressBarX
            <DebuggerNonUserCode> _
            Get
                Return Me._prgMain
            End Get
            <MethodImpl(MethodImplOptions.Synchronized), DebuggerNonUserCode> _
            Set(ByVal WithEventsValue As ProgressBarX)
                Me._prgMain = WithEventsValue
            End Set
        End Property

        Friend Overridable Property prgAlternate As ProgressBarX
            <DebuggerNonUserCode> _
            Get
                Return Me._prgAlternate
            End Get
            <MethodImpl(MethodImplOptions.Synchronized), DebuggerNonUserCode> _
            Set(ByVal WithEventsValue As ProgressBarX)
                Me._prgAlternate = WithEventsValue
            End Set
        End Property

        Friend Overridable Property TableLayoutPanel1 As TableLayoutPanel
            <DebuggerNonUserCode> _
            Get
                Return Me._TableLayoutPanel1
            End Get
            <MethodImpl(MethodImplOptions.Synchronized), DebuggerNonUserCode> _
            Set(ByVal WithEventsValue As TableLayoutPanel)
                Me._TableLayoutPanel1 = WithEventsValue
            End Set
        End Property

        Friend Overridable Property lblFile As LabelX
            <DebuggerNonUserCode> _
            Get
                Return Me._lblFile
            End Get
            <MethodImpl(MethodImplOptions.Synchronized), DebuggerNonUserCode> _
            Set(ByVal WithEventsValue As LabelX)
                Me._lblFile = WithEventsValue
            End Set
        End Property

        Friend Overridable Property lblJob As LabelX
            <DebuggerNonUserCode> _
            Get
                Return Me._lblJob
            End Get
            <MethodImpl(MethodImplOptions.Synchronized), DebuggerNonUserCode> _
            Set(ByVal WithEventsValue As LabelX)
                Me._lblJob = WithEventsValue
            End Set
        End Property

        Private Property _SourceDir As String
            <DebuggerNonUserCode> _
            Get
                Return Me.__SourceDir
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As String)
                Me.__SourceDir = AutoPropertyValue
            End Set
        End Property

        Private Property _OutputName As String
            <DebuggerNonUserCode> _
            Get
                Return Me.__OutputName
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As String)
                Me.__OutputName = AutoPropertyValue
            End Set
        End Property

        Private Property _BytesTransfered As Long
            <DebuggerNonUserCode> _
            Get
                Return Me.__BytesTransfered
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As Long)
                Me.__BytesTransfered = AutoPropertyValue
            End Set
        End Property

        Private Property _JobSize As Long
            <DebuggerNonUserCode> _
            Get
                Return Me.__JobSize
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As Long)
                Me.__JobSize = AutoPropertyValue
            End Set
        End Property

        Private Property _MaxFileSize As Long
            <DebuggerNonUserCode> _
            Get
                Return Me.__MaxFileSize
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As Long)
                Me.__MaxFileSize = AutoPropertyValue
            End Set
        End Property

        Private Property _DeleteFilesAfterZip As Boolean
            <DebuggerNonUserCode> _
            Get
                Return Me.__DeleteFilesAfterZip
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As Boolean)
                Me.__DeleteFilesAfterZip = AutoPropertyValue
            End Set
        End Property

        Private Property _HideForm As Boolean
            <DebuggerNonUserCode> _
            Get
                Return Me.__HideForm
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As Boolean)
                Me.__HideForm = AutoPropertyValue
            End Set
        End Property

        Public Property MaxFileSize As Long
            Get
                Return Me._MaxFileSize
            End Get
            Set(ByVal value As Long)
                Me._MaxFileSize = ((value * &H400) * &H400)
            End Set
        End Property

        Public Property DeleteFilesAfterZip As Boolean
            Get
                Return Me._DeleteFilesAfterZip
            End Get
            Set(ByVal value As Boolean)
                Me._DeleteFilesAfterZip = value
            End Set
        End Property

        Public Property HideForm As Boolean
            Get
                Return Me._HideForm
            End Get
            Set(ByVal value As Boolean)
                Me._HideForm = value
            End Set
        End Property


        ' Fields
        Private Shared __ENCList As List(Of WeakReference) = New List(Of WeakReference)
        Private components As IContainer
        <AccessedThroughProperty("prgMain")> _
        Private _prgMain As ProgressBarX
        <AccessedThroughProperty("prgAlternate")> _
        Private _prgAlternate As ProgressBarX
        <AccessedThroughProperty("TableLayoutPanel1")> _
        Private _TableLayoutPanel1 As TableLayoutPanel
        <AccessedThroughProperty("lblFile")> _
        Private _lblFile As LabelX
        <AccessedThroughProperty("lblJob")> _
        Private _lblJob As LabelX
        <DebuggerBrowsable(DebuggerBrowsableState.Never), CompilerGenerated> _
        Private __SourceDir As String
        <DebuggerBrowsable(DebuggerBrowsableState.Never), CompilerGenerated> _
        Private __OutputName As String
        <DebuggerBrowsable(DebuggerBrowsableState.Never), CompilerGenerated> _
        Private __BytesTransfered As Long
        <CompilerGenerated, DebuggerBrowsable(DebuggerBrowsableState.Never)> _
        Private __JobSize As Long
        <CompilerGenerated, DebuggerBrowsable(DebuggerBrowsableState.Never)> _
        Private __MaxFileSize As Long
        <DebuggerBrowsable(DebuggerBrowsableState.Never), CompilerGenerated> _
        Private __DeleteFilesAfterZip As Boolean
        <CompilerGenerated, DebuggerBrowsable(DebuggerBrowsableState.Never)> _
        Private __HideForm As Boolean

        ' Nested Types
        Private Delegate Sub SaveEntryProgress(ByVal e As SaveProgressEventArgs)
    End Class
End Namespace

